proxy/src/ctx/transport.rs

```diff
@@ -36,6 +51,13 @@ impl Ctx {
             Ctx::Server(ref ctx) => &ctx.proxy,
         }
     }
+
+    pub fn tls_status(&self) -> TlsStatus {
+        match *self {
+            Ctx::Client(ref ctx) => ctx.tls_status,
+            Ctx::Server(ref ctx) => ctx.tls_status,
```

Comment:
You can combine these: `Ctx::Client(ctx) | Ctx::Server(ctx) => ctx.tls_status`.

Also you can drop the `ref`s if you drop the `*` in `*self`.